services/brig/test/integration/API/User/Auth.hs

```diff
@@ -237,6 +230,46 @@ testThrottleLogins conf b = do
         threadDelay (1000000 * (n + 1))
     void $ login b (defEmailLogin e) SessionCookie
 
+-------------------------------------------------------------------------------
+-- Backdoor login
+
+-- | Check that standard email login works with @/backdoor-login@ even
+-- without having the right password.
+testEmailBackdoorLogin :: Brig -> Http ()
+testEmailBackdoorLogin brig = do
+    u <- randomUser brig
+    let Just email = userEmail u
+    now <- liftIO getCurrentTime
+
+    -- Login with email and do some checks
+    let loginSpec = BackdoorLogin (LoginByEmail email) Nothing
+    _rs <- backdoorLogin brig loginSpec PersistentCookie
+        <!! const 200 === statusCode
+    liftIO $ do
+        assertSanePersistentCookie (decodeCookie _rs)
+        assertSaneAccessToken now u (decodeToken _rs)
+
+-- | Check that @/backdoor-login@ can not be used to login as a suspended
+-- user.
+testSuspendedBackdoorLogin :: Brig -> Http ()
+testSuspendedBackdoorLogin brig = do
+    -- Create a user and immediately suspend them
+    u <- randomUser brig
+    let Just email = userEmail u
+    setStatus brig (userId u) Suspended
+    -- Try to login and see if we fail
+    let loginSpec = BackdoorLogin (LoginByEmail email) Nothing
+    backdoorLogin brig loginSpec PersistentCookie
+        !!! const 403 === statusCode
+
+-- | Check that @/backdoor-login@ fails if the user doesn't exist.
+testNoUserBackdoorLogin :: Brig -> Http ()
+testNoUserBackdoorLogin brig = do
+    -- Try to login as wrong@wire.com and see if we fail
+    let loginSpec = BackdoorLogin (LoginByEmail (Email "wrong" "wire.com")) Nothing
+    backdoorLogin brig loginSpec PersistentCookie
+        !!! const 403 === statusCode
```

Comment:
we need to have a way of distinguishing `user does not exist` from `something else happened`, so that we can decide whether to create the user and try again.  this should also be tested here.

Reply:
You can do that by inspecting the `"label"` in the response. I added explicit tests for that.

Reply:
:+1: